Regrouper la documentation dans ce dépôt

thom4parisot opened the pull request:
Actuellement la documentation utilisateur est dans un dépôt séparé : https://github.com/EcrituresNumeriques/stylo-doc

Elle est publiée via GitHub Pages à l'adresse : https://stylo-doc.ecrituresnumeriques.ca/fr_FR/#!index.md

Peut être que nous pourrions regrouper la documentation dans le répertoire `docs`. 

Pour la publication, nous pouvons rester sur GitHub Pages et MDwiki ou tester d'autres alternatives comme [Docusaurus](https://github.com/facebook/docusaurus) et Netlify...

exalted committed:
Add a good default .gitattributes for web development

exalted committed:
Add custom attributes for mdwiki & github pages

exalted committed:
Add attributes for documents and even more graphics

exalted committed:
Cosmetic changes

exalted committed:
Add good default ignore rules

exalted committed:
Add 0.6.2 version of MDwiki

… along with a README that explains briefly how to update to a later version in the future.

exalted committed:
Add placeholder content for the home page

And add some information into README that explains how to preview changes locally.

exalted committed:
Change structure so that it allows to create more than one wiki

… for each language that you would like to add.

exalted committed:
Small change in a comment

exalted committed:
Explain in detail how to create your own wiki

exalted committed:
Explain "fork it" in README ;-)

exalted committed:
Change structure for uploads folder

And small small changes in the README.

teopost committed:
Added Linux alternative to Pow

exalted committed:
Merge pull request #1 from teopost/patch-1

Added Linux alternative to Pow

exalted committed:
Explain how to create one than one wiki

… using the same GitHub user or organization account

exalted committed:
Improved README for "cloning" a little.

exalted committed:
Better "clone" instructions for the CLI

exalted committed:
Small small changes in the README.

exalted committed:
Add `title` to `config.json` which will set `html>head>title`

Read more at https://github.com/Dynalon/mdwiki/issues/58#issuecomment-29785288

--
Close #4

exalted committed:
Cosmetic changes

bplmp committed:
Update 'How to Preview' section for Macs

Suggest using python simpleserver to serve static site on Mac.

bplmp committed:
Fix typos

exalted committed:
Merge pull request #8 from bplmp/patch-1

Update 'How to Preview' section for Macs

exalted committed:
Fix small typo

exalted committed:
Fix small typo

Commit:
Gave some :heart: to windows section

Added Fenix Web Server as the static web host for Windows.

Commit:
Merge pull request #1 from theTechnoKid/patch-1

Gave some :heart: to windows section

exalted committed:
Merge pull request #9 from theTechnoKid/gh-pages

Gave some :heart: to Windows section.

lakonis committed:
initialisation de la doc

PookMook committed:
Update index.md

lakonis committed:
Merge pull request #1 from PookMook/patch-1

Update index.md

lakonis committed:
Create CNAME

marviro committed:
première ébauche doc

marviro committed:
ajout about et contacts

marviro committed:
ajout titre du wiki en français

lakonis committed:
full update

lakonis committed:
update, add button and license cc

lakonis committed:
ajout biblio, metadata et nv modal versions

marviro committed:
correction lien premiers pas

marviro committed:
jout section statistiqes

marviro committed:
ajout infos sur le balisage md

marviro committed:
correction lien navigation

marviro committed:
documentation annotation

lakonis committed:
update juin premierspas

lakonis committed:
Merge branch 'gh-pages' into nicolas
Fait converger la version marcello (juillet) avec modifs nicolas (juin non commites)

lakonis committed:
premierpas - corrections post-merge

antoinentl committed:
edit: ajout de l'indication de balisage des liens FR

antoinentl committed:
edit: ajout de l'indication de balisage des citations FR

antoinentl committed:
edit: ajout de l'indication de balisage des images FR

marviro committed:
Merge pull request #2 from antoinentl/gh-pages

Balisage des liens, citations et images

lakonis committed:
Doc sur l'espace insécable et raccourci clavier

marviro committed:
ajout vidéo démo

lakonis committed:
add doc bibliographie

lakonis committed:
update exports

lakonis committed:
update bibliographie

lakonis committed:
add publications

lakonis committed:
ajout bouton demo sur index fr

lakonis committed:
restructuration de la doc

Commit:
changements page about + premierpas

Commit:
syntaxe md- ajouts

Commit:
ajout export preview

Commit:
ajout infos export

lakonis committed:
amelioration balisage semantique et tableau

Mmellet committed:
début corrections premiers pas

lakonis committed:
Merge branch 'gh-pages' of https://github.com/EcrituresNumeriques/stylo-doc into gh-pages

Mmellet committed:
Merge branch 'gh-pages' of https://github.com/EcrituresNumeriques/stylo-doc into gh-pages

Mmellet committed:
Fini les premiers pas

Mmellet committed:
ajout premiers pas

Mmellet committed:
Mise à jour

Mmellet committed:
MIse à jour

Mmellet committed:
Mise à jour images

Mmellet committed:
Mise à jour

lakonis committed:
publications: ajout du colloque numerev

lakonis committed:
premierspas: quelques modifs, rajout de la structure de l'article

lakonis committed:
premierspas: correction lien interne

Mmellet committed:
ajout images

antoinentl committed:
edit: ajout d'exemples de balisage RDFa

antoinentl committed:
Merge pull request #3 from antoinentl/gh-pages

Ajout d'exemples de balisage RDFa

Mmellet committed:
communication sur permanences stylo

antoinentl committed:
edit: permanence Stylo sur la home

antoinentl committed:
Merge pull request #4 from antoinentl/gh-pages

Permanence Stylo sur la home

antoinentl committed:
fix: permanence sur la home en taille acceptable

antoinentl committed:
Merge pull request #5 from antoinentl/gh-pages

fix: permanence sur la home en taille acceptable

antoinentl committed:
fix: suppression d'un HR superficiel

antoinentl committed:
Merge pull request #6 from antoinentl/gh-pages

fix: suppression d'un HR superficiel

antoinentl committed:
edit: mail, francisation, GitHub

antoinentl committed:
fix: mise en commentaire des entrées vides

antoinentl committed:
fix: coquille

antoinentl committed:
fix: formulations, liens

antoinentl committed:
new: page gérer mes articles

antoinentl committed:
fix: présentation de la documentation

antoinentl committed:
Merge pull request #7 from antoinentl/gh-pages

Mises à jour de la documentation

Mmellet committed:
Ajout images dans doc-fr

antoinentl committed:
fix: suppression des entrées non utilisées

antoinentl committed:
fix: lien et suppression mention

antoinentl committed:
fix: lien et suppression mention

antoinentl committed:
Merge pull request #8 from antoinentl/gh-pages

Pied de page et menu

antoinentl committed:
fix: Chaire et non lien

antoinentl committed:
fix: lien cassé et import Zotero

antoinentl committed:
fix: corrections formulations

antoinentl committed:
edit: relecture

antoinentl committed:
fix: lien numerev et permanence

antoinentl committed:
Merge pull request #9 from antoinentl/gh-pages

Relecture LN

Mmellet committed:
édition de la page Gérer mes articles

Mmellet committed:
Modif tableaux Mesarticles.md

Mmellet committed:
suppression page export.md

antoinentl committed:
fix: suppression de l'entrée export dans le menu

antoinentl committed:
edit: ajout d'une publication

antoinentl committed:
Merge pull request #10 from antoinentl/gh-pages

Suppression export et ajout publication

Mmellet committed:
Ajout import et modif mineures

Mmellet committed:
ajout import dans premierspas.md

Mmellet committed:
Merge branch 'gh-pages' of https://github.com/EcrituresNumeriques/stylo-doc into gh-pages

Mmellet committed:
Modif note

Mmellet committed:
Modif note 2

Mmellet committed:
Mise à jour liens

antoinentl committed:
edit: coquille, lien vers vision, etc.

antoinentl committed:
edit: coquille, lien vers vision, etc.

Mmellet committed:
Annotation

Mmellet committed:
Corrections mineures

Mmellet committed:
ajout allCitation.png

Mmellet committed:
all citation option dans Métadonnées

antoinentl committed:
edit: notes et relecture syntaxe Markdown

antoinentl committed:
fix: affichage tableau

Mmellet committed:
Creation page

antoinentl committed:
Merge branch 'gh-pages' of github.com:antoinentl/stylo-doc into gh-pages

Mmellet committed:
ajout page résoudre problèmes

antoinentl committed:
Merge pull request #11 from antoinentl/gh-pages

Amélioration affichage, corrections coquilles

antoinentl committed:
fix: affichage des tableaux

antoinentl committed:
fix: mdwiki ne gère pas les footnotes ☹

antoinentl committed:
Merge pull request #12 from antoinentl/gh-pages

Fix: tableaux et footnotes

antoinentl committed:
fix: liens footnotes mi-relatifs, mi-absolus

antoinentl committed:
Merge pull request #13 from antoinentl/gh-pages

fix: liens footnotes mi-relatifs, mi-absolus

Mmellet committed:
Edit About

Mmellet committed:
Mise à jour

Mmellet committed:
ajout images pour doc en

Mmellet committed:
Premier draft

Mmellet committed:
Mise à jour

Mmellet committed:
format md

Mmellet committed:
tableau correction

Mmellet committed:
Update firststeps.md

Mmellet committed:
Premier draft

Mmellet committed:
correction mineure

Mmellet committed:
premier draft

Mmellet committed:
Update managingref.md

Mmellet committed:
Update managingref.md

antoinentl committed:
Page de documentation des mémoires et thèses, aussi appelé Books

antoinentl committed:
Merge pull request #14 from antoinentl/gh-pages

Page de documentation des mémoires et thèses, aussi appelé Books

antoinentl committed:
fix: chemin des deux images

antoinentl committed:
Merge pull request #15 from antoinentl/gh-pages

fix: chemin des deux images

thom4parisot commented:
Sachant que la documentation peut être republiée telle quelle, sans changement d'outil technique (MDWiki). Elle serait lisible sur https://ecrituresnumeriques.github.io/stylo, et on peut tout à fait faire en sorte que `stylo-doc.ecrituresnumeriques.ca` pointe dessus.

@antoinentl, y'a une raison pour que les dépôts soient séparés ? C'est une documentation utilisateur ? Ou … ?

marviro commented:
On peut les mettre ensemble - il suffit de créer une section documentation technique. Comme vous préférez. Voulez-vous que je vous donne accès?

antoinentl commented:
Pour répondre à tes questions @oncletom :

- oui il s'agit d'une documentation utilisateur ;
- prévoir un volet développeur pourrait être cohérent et simplifierait la lisibilité du projet (et si on doit déplacer des choses ça fera moins de choses à déplacer :+1: ).

Je vous laisse vous charger de cela ou vous voulez que je prenne ce point en main @oncletom @Mogztter ?

thom4parisot commented:
Super, merci pour vos réponses.

On regardera à prioriser ça pour avril & cie, histoire de garder le focus sur l'intégration avec Human-ID. Dans tous les cas, vu que c'est un dépôt Git, réintégrer les contenus se fera facilement — je verrai si on peut conserver l'historique sans trop de difficultés.

oncletom committed:
Add 'docs/' from commit '3c522310d374699f775af5d0d898c7b301222699'

git-subtree-dir: docs
git-subtree-mainline: 35dbbdca9ae413d440a0f7904d153af4072adb05
git-subtree-split: 3c522310d374699f775af5d0d898c7b301222699

thom4parisot commented:
Réalisé avec la commande suivante 

```bash
git subtree add --prefix docs https://github.com/EcrituresNumeriques/stylo-doc.git gh-pages
```

antoinentl commented:
@oncletom Tu peux me confirmer ce qui va se passer une fois la PR validée ? La documentation sera intégrée dans une branche isolée ? Pas de risque de _polluer_ l'historique du dépôt Stylo entre le code et la documentation ? Merci pour tes éclaircissements !

thom4parisot commented:
Une fois la PR mergée, un répertoire `docs` va être créé, avec toute la doc présente dans la branche `gh-pages` de https://github.com/EcrituresNumeriques/stylo, et ça ramène tout l'historique avec.

C'est aussi possible de ramener tout le contenu, sans l'historique — tu me dis et je peux faire la manip.

Explications complémentaires sur https://www.atlassian.com/git/tutorials/git-subtree (ce qui entrouvre aussi le fait de brancher les templates en se passant des submodules).

antoinentl commented:
@oncletom Merci pour tes précisions !

ggrossetie commented:
Top merci Thomas :+1:

thom4parisot commented:
@antoinentl top !

La doc est aujourd'hui accessible sur https://ecrituresnumeriques.github.io/stylo/ (et donc https://ecrituresnumeriques.github.io/stylo/fr_FR/ etc.)

Si le CNAME est remis à la racine du site, le nom de domaine `stylo-doc.ecrituresnumeriques.ca` y sera associé (si ce CNAME est viré du dépôt `stylo-doc`, qui peut être archivé au passage).

thom4parisot commented:
En fait, y'aurait juste à supprimer le CNAME du dépôt `stylo-doc`, et il n'y a rien d'autre à changer.
